--- model1.stan	2022-12-23 09:42:25.359985957 +0900
+++ model2.stan	2022-12-23 09:42:25.359985957 +0900
@@ -1,4 +1,4 @@
-// model1.stan
+// model2.stan
 
 // function to avoid 0 and 1
 functions{
@@ -28,6 +28,9 @@
   real<lower=0> s_mu0[ND];
   real<lower=0> log_s_sigma0[ND];
 
+  // offset value for each individual
+  real d[NG];
+
   // parameters of each drug and animal
   real mu[ND, NG];
   real<lower=0> sigma[ND, NG];
@@ -49,9 +52,10 @@
 
 model {
   // parameter of each drug (each animal)
+  // with offset value for each animal
   for (i in 1:ND) {
     for (j in 1:NG) {
-      mu[i, j] ~ normal(mu0[i], s_mu0[i]);
+      mu[i, j] ~ normal(mu0[i] + d[j], s_mu0[i]);
       sigma[i, j] ~ lognormal(log_sigma0[i], log_s_sigma0[i]);
     }
   }
@@ -59,6 +63,7 @@
   mu0 ~ cauchy(50, 20);
   s_mu0 ~ cauchy(0, 1);  // halfCauchy
   log_sigma0 ~ normal(2.5, 1);
+  d ~ normal(0, 20);
 
   SCORE ~ binomial(6, p);
 }
